test/parallel/test-readable-single-end.js

```diff
@@ -0,0 +1,16 @@
+'use strict';
+
+const common = require('../common');
+const { Readable } = require('stream');
+
+// This test ensures that there will not be an additional empty 'readable'
+// event when stream has ended (only 1 event signalling about end)
+
+const r = new Readable({
+  read: () => {},
+});
+
+r.push(null);
+
+r.once('readable', common.mustCall());
+r.once('end', common.mustCall());
```

Comment:
When reading the description of the test it seems like these should be `.on` instead of once?